lib/tukey/data_set.rb

```diff
@@ -127,6 +127,7 @@ def leaf_labels
   #
   def filter(leaf_label_id = nil, keep_leafs: false, orphan_strategy: :destroy, &block)
     fail ArgumentError, 'No block and no leaf_label_id passed' if !block_given? && leaf_label_id.nil?
+    return self.dup if self.data.empty?
```

Comment:
Dit gaat stuk als je `filter` aanroept op een value dataset (bijv. `DataSet.new(data: 4)`). Nu ging dat door de `self.data.each_with_object` hieronder sowieso al stuk. Je zou eventueel nog een `fail "Cannot filter on value DataSets" unless data_array?` kunnen toevoegen. Kijk maar of je daar nog zin in hebt.